dotCMS/src/main/java/com/dotcms/rest/api/v1/container/ContainerResource.java

```diff
@@ -92,24 +120,244 @@ public ContainerResource(final WebResource webResource,
+                                mode == PageMode.ANON);
+
+
+        APILocator.getPermissionAPI().checkPermission(contentlet, PermissionLevel.EDIT, user);
+        APILocator.getPermissionAPI().checkPermission(container, PermissionLevel.EDIT, user);
+
+
+
+        MultiTree mt =
+                new MultiTree().setContainer(containerId).setContentlet(contentletId).setRelationType(uid).setTreeOrder(order);
+
+        MultiTreeFactory.saveMultiTree(mt);
+
+
+
+        return Response.ok("ok").build();
+    }
+
+
+    @DELETE
+    @JSONP
+    @NoCache
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    @Path("delete/{containerId}/content/{contentletId}/uid/{uid}")
+    public final Response removeContentletFromContainer(@Context final HttpServletRequest req,
+            @Context final HttpServletResponse res, @PathParam("containerId") final String containerId,
+            @PathParam("contentletId") final String contentletId, @QueryParam("order") final long order,
+            @PathParam("uid") final String uid) throws DotDataException, DotSecurityException, ParseErrorException,
+            MethodInvocationException, ResourceNotFoundException, IOException, IllegalAccessException, InstantiationException,
+            InvocationTargetException, NoSuchMethodException {
+
+        final InitDataObject initData = webResource.init(true, req, true);
+        final User user = initData.getUser();
+        final PageMode mode = PageMode.get(req);
+        final Language id = WebAPILocator.getLanguageWebAPI().getLanguage(req);
+
+
+        
+        
+        
+        
+        
+        
+
+        return Response.ok("ok").build();
+    }
+
+
+
+    @Path("/containerContent/{params:.*}")
+    public final Response containerContents(@Context final HttpServletRequest req, @Context final HttpServletResponse res,
+            @QueryParam("containerId") final String containerId, @QueryParam("contentInode") final String contentInode)
+            throws DotDataException, DotSecurityException, ParseErrorException, MethodInvocationException,
+            ResourceNotFoundException, IOException {
+
+        final InitDataObject initData = webResource.init(true, req, true);
+        final User user = initData.getUser();
+
+        Language id = WebAPILocator.getLanguageWebAPI().getLanguage(req);
+
+        PageMode mode = PageMode.get(req);
+
+        Container container = APILocator.getContainerAPI().find(containerId, user, mode == PageMode.ANON);
+
+        org.apache.velocity.context.Context context = VelocityUtil.getWebContext(req, res);
+        Contentlet contentlet = APILocator.getContentletAPI().find(contentInode, user, mode == PageMode.ANON);
+        ContainerStructure cStruct = APILocator.getContainerAPI().getContainerStructures(container).stream()
+                .filter(cs -> contentlet.getStructureInode().equals(cs.getStructureId())).findFirst().orElse(null);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Specify a type for: 'cs' [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2211)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "getStructureInode"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)